dynatrace/src/test/java/com/dynatrace/opentelemetry/metric/InMemoryDtMetricReader.java

```diff
@@ -0,0 +1,46 @@
+package com.dynatrace.opentelemetry.metric;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import io.opentelemetry.sdk.metrics.export.MetricReaderFactory;
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Inspired on
+ * https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/testing/InMemoryMetricReader.java
+ * A reader that calls the exporter whenever it receives a flush call from the MeterProvider. Useful
+ * for testing the exporter working with a configured MeterProvider
+ */
+class InMemoryDtMetricReader implements MetricReader, MetricReaderFactory {
```

Comment:
The InMemory from OTel doesn't call our exporter and simply expose a way to collect the metrics. Since I wanted to test our exporter with the MeterProvider this was a way I find out.

I agree with the DT, will change it to something else :)